backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/CloneVmCommand.java

```diff
@@ -385,21 +400,23 @@ protected Guid getStoragePoolIdFromSourceImageContainer() {
      */
     @Override
     protected void lockEntities() {
-        TransactionSupport.executeInNewTransaction(() -> {
-            Snapshot snapshot = snapshotDao.get(getParameters().getSourceSnapshotId(), null, false);
-            getCompensationContext().snapshotEntityStatus(snapshot);
-            snapshotDao.updateStatus(snapshot.getId(), Snapshot.SnapshotStatus.LOCKED);
-            getCompensationContext().stateChanged();
-            return null;
-        });
-        freeLock();
+        // No need to lock snapshots for VMs with MBS disks
+        if (isNoManagedBlockDiskAttachedToVm()) {
+            TransactionSupport.executeInNewTransaction(() -> {
+                Snapshot snapshot = snapshotDao.get(getParameters().getSourceSnapshotId(), null, false);
+                getCompensationContext().snapshotEntityStatus(snapshot);
+                snapshotDao.updateStatus(snapshot.getId(), Snapshot.SnapshotStatus.LOCKED);
+                getCompensationContext().stateChanged();
+                return null;
+            });
+            freeLock();
+        }
     }
 
     @Override
     protected void unlockEntities() {
         // Assumption - this is last DB change of command, no need for compensation here
         snapshotDao.updateStatus(getParameters().getSourceSnapshotId(), Snapshot.SnapshotStatus.OK);
-        vmDynamicDao.updateStatus(getVmId(), VMStatus.Down);
```

Comment:
Are you sure this is not needed?

Reply:
After re-testing, this one is needed when the VM is running.
For non-running VMs, the status remains '0' during the while command.
Will add it back.